diff --git a/direnvrc b/direnvrc
index 723f29c..ee3668f 100644
--- a/direnvrc
+++ b/direnvrc
@@ -243,12 +243,18 @@ use_flake() {
     else
       _nix_clean_old_gcroots "$layout_dir"
 
-      # We need to update our cache
-      local tmp_profile="${layout_dir}/flake-profile.$$"
-      local tmp_profile_rc
-      tmp_profile_rc=$("${NIX_BIN_PREFIX}nix" print-dev-env \
+    # We need to update our cache
+    local tmp_profile="${layout_dir}/flake-profile.$$"
+    local tmp_profile_rc
+      
+      "${NIX_BIN_PREFIX}nix" print-dev-env \
         --extra-experimental-features "nix-command flakes" \
-        --profile "$tmp_profile" "$@")
+        --profile "$tmp_profile" "$@y" \
+        --log-format internal-json -v |& nom --json 
+ 
+    tmp_profile_rc=$("${NIX_BIN_PREFIX}nix" print-dev-env \
+      --extra-experimental-features "nix-command flakes" \
+      --profile "$tmp_profile" "$@")
 
       local drv
       drv=$(_nix_direnv_realpath "$tmp_profile")
